### ISSUE TITLE
Allow sending in an object to the {{ meta_tags }} tag, for integration with non-page types

### ISSUE THREAD

**Issue by timheap** [takeflight/wagtail-metadata#6]
For views that display non-page models, the ability to supply a non-page model to `{% meta_tags %}` would be very valuable:

```django+html
{% meta_tags foo %}
```

`wagtailnews` would also benefit greatly from this, as you could have customised settings on a per-post basis. This could allow for something like:

```django+html
{% meta_tags newsitem page %}
```

to pull the image/description from either the current news item, the page, or the global settings, depending upon what had been populated.

To assist developers in matching the required model API, you could provide a MetadataMixin model, much like the MetadataPageMixin, but with the addition of a `search_description` field and no preconfigured panels (who knows if the model is even editable through the wagtail admin, after all...):

```python
class MetadataMixin(models.Model):
    search_image = models.ForeignKey('wagtailimages.Image', null=True, blank=True, on_delete=models.SET_NULL)
    search_description = models.TextField(
       help_text=("This is the description displayed when this item is "
       " shared on social media or is found on search engines")
```

**Comment by timheap** [comment]
Actually, come talk to me about this. I have some thoughts...

**Comment by timheap** [comment]
Closed in #21
